tools/wpt/browser.py

```diff
@@ -17,33 +17,46 @@
 
 uname = platform.uname()
 
-def path(path, exe):
-    path = path.replace("/", os.path.sep)
-    if exe and uname[0] == "Windows":
-        path += ".exe"
-    return path
-
 
 class Browser(object):
     __metaclass__ = ABCMeta
 
     @abstractmethod
     def install(self, dest=None):
+        """Install the browser."""
+        return NotImplemented
+
+    @abstractmethod
+    def install_webdriver(self, dest=None):
+        """Install the WebDriver implementation for this browser."""
+        return NotImplemented
+
+    @abstractmethod
+    def find_binary(self):
+        """Find the binary of the browser.
+
+        If the WebDriver for the browser is able to find the binary itself, this
+        method doesn't need to be implemented, in which case NotImplementedError
+        is suggested to be raised to prevent accidental use.
+        """
         return NotImplemented
 
     @abstractmethod
-    def install_webdriver(self):
+    def find_webdriver(self):
+        """Find the binary of the WebDriver."""
         return NotImplemented
 
     @abstractmethod
-    def version(self):
+    def version(self, root):
+        """Retrieve the release version of the installed browser."""
         return NotImplemented
 
     @abstractmethod
     def requirements(self):
         """Name of the browser-specific wptrunner requirements file"""
         return NotImplemented
 
+    # TODO(robertma): This method doesn't seem to be called anywhere.
```

Comment:
This was used to work around a Chrome issue, but I think it's dead code now.

Reply:
Dead code has been removed.